### PR TITLE
fix: EXPOSED-16 Failed tests in KotlinTimeTests

### PR DESCRIPTION
**Pull request by joc-a**
The same tests pass for SQLite in JavaTimeTests because the comparison is done differently for Java. A Temporal comparison function is used for Java, but not for Kotlin. This commit changes the Kotlin tests to match those for Java.

### CODE REVIEW

File: exposed-kotlin-datetime/src/test/kotlin/org/jetbrains/exposed/sql/kotlin/datetime/KotlinTimeTests.kt [hunk 1]
```diff
@@ -91,15 +91,33 @@ open class KotlinTimeBaseTest : DatabaseTestsBase() {
...
     when {
         d1 == null && d2 == null -> return
         d1 == null -> error("d1 is null while d2 is not on ${currentDialectTest.name}")
         d2 == null -> error("d1 is not null while d2 is null on ${currentDialectTest.name}")
-        else -> {
-            assertEquals(d1.toJavaLocalDateTime().toEpochSecond(ZoneOffset.UTC), d2.toJavaLocalDateTime().toEpochSecond(ZoneOffset.UTC), "Failed on epoch seconds ${currentDialectTest.name}")
+        d1 is LocalTime && d2 is LocalTime -> {
+            assertEquals(d1.toSecondOfDay(), d2.toSecondOfDay(), "Failed on seconds ${currentDialectTest.name}")
+            if (d2.nanosecond != 0) {
+                assertEqualFractionalPart(d1.nanosecond, d2.nanosecond)
```
Comment (e5l) [inline exposed-kotlin-datetime/src/test/kotlin/org/jetbrains/exposed/sql/kotlin/datetime/KotlinTimeTests.kt:102]: I would also add a message hrere

**Review by e5l — APPROVED** [review]
lgtm

File: exposed-kotlin-datetime/src/test/kotlin/org/jetbrains/exposed/sql/kotlin/datetime/KotlinTimeTests.kt (hunk 1 above)
Comment (joc-a) [reply exposed-kotlin-datetime/src/test/kotlin/org/jetbrains/exposed/sql/kotlin/datetime/KotlinTimeTests.kt:102]: @e5l As a parameter to `assertEqualFractionalPart` function you mean?

Comment (e5l) [reply exposed-kotlin-datetime/src/test/kotlin/org/jetbrains/exposed/sql/kotlin/datetime/KotlinTimeTests.kt:102]: yep

**Review by e5l — APPROVED** [review]

File: exposed-kotlin-datetime/src/test/kotlin/org/jetbrains/exposed/sql/kotlin/datetime/KotlinTimeTests.kt (hunk 1 above)
Comment (joc-a) [reply exposed-kotlin-datetime/src/test/kotlin/org/jetbrains/exposed/sql/kotlin/datetime/KotlinTimeTests.kt:102]: @e5l Every case (for every DB dialect) within that function already has its own message. What other message would we add?

**Pull request merged**